libretro-common/include/libretro.h

```diff
@@ -976,6 +976,19 @@ struct retro_hw_render_context_negotiation_interface
                                             * so it will be used after SET_HW_RENDER, but before the context_reset callback.
                                             */
 
+struct retro_resource
+{
+   const char *rel_path;
+   const char *base_path;
```

Comment:
Oh, you mean like that.

Personally I'd prefer deallocating on the next call to that env, or next call to any env whatsoever, or something. Core can then strdup whatever it wants.

Or creating a virtual filesystem layer. Front could merge that from multiple directories.